### ISSUE TITLE
Document.pid mapping adds the prefix pid

### ISSUE THREAD

**Issue by antolinos** [ral-facilities/datagateway-api#356]
It is more a question than an issue.  Why is it prefixing the field `pid` with `pid:` when value is a integer?

### Description:
I am using the default `search_api_mapping.json` mapping file where the `pid` of a document is defined as:
```
  "Document": {
    "base_icat_entity": "Investigation",
    "pid": ["doi", "id"],
```
If I do:
```
http://127.0.0.1:5000/search-api/documents
```

The returned pid  values will be look like:
```
pid | "10.15151/ESRF-ES-94520246"
pid | "pid:13175377"
```
I see that it is not an error as it is forced in the code:
https://github.com/ral-facilities/datagateway-api/blob/c67420548dd4cf951c31195362557496880ce7bb/datagateway_api/src/search_api/models.py#L253

However, by looking at the `Document` definition given in the search-api impementation for sci-cat, I did not see that it was mandatory:
https://github.com/panosc-eu/search-api/blob/master/doc/data-model.md#document

So, I was wondering the reason why there is a prefix.

Thanks!

**Comment by VKTB** [comment]
Hi @antolinos, thank you for your question.

This is not mandatory as per the Data Model guide but it is a best attempt solution to make the identifiers persistent. The corresponding ICAT columns for the `pid` PaNOSC fields are nullable (optional), meaning that the values of these columns for some entity instances could be set to `NULL` in ICAT. In the case of ISIS, we discovered that the `pid` values of some ICAT Samples are set to `NULL` and this is when we decided that when such case occurs that its `id` is instead taken and used in setting the value of the PaNOSC `pid` field in the `pid:<id>` format. During this stage, we also discovered that the other ICAT entities have nullable corresponding `pid` columns. Because of this, we decided to apply the same solution to the rest of the PaNOSC entities, not just `Sample`. The decision to prefix the `pid` at the Search API level was based on the suggestions in https://github.com/icatproject/icat.server/issues/231.

Considering the above, your Document with pid `pid:13175377` is prefixed with `pid:` because the `doi` value of the corresponding investigation in ICAT is `NULL`, therefore it takes its `id` (the alternative ICAT mapping field) and sets the `pid` to be in the `pid:<id>` format. I hope this makes sense.

Relevant issue to the work mentioned above - https://github.com/ral-facilities/datagateway-api/issues/314
Relevant PR to the work mentioned above - https://github.com/ral-facilities/datagateway-api/pull/324
